File: clients/android/AwareChat-Android/app/src/main/java/com/example/awarechat_android/core/network/ApiEndpoint.kt
package com.example.awarechat_android.core.network

import okhttp3.Request

enum class ApiEndpoint(private val pathSegment: String) {
    Health("health"),
    Users("users");

    fun request(configuration: NetworkConfiguration): Request = Request.Builder()
        .url(configuration.httpBaseUrl.newBuilder().addPathSegment(pathSegment).build())
        .get()
        .header("Accept", "application/json")
        .build()
}

File: clients/ios/AwareChat-iOS/AwareChat-iOS/Core/Networking/APIEndpoint.swift
import Foundation

nonisolated enum APIEndpoint: String, CaseIterable, Sendable {
  case health
  case users

  func request(configuration: NetworkConfiguration) -> URLRequest {
    var request = URLRequest(
      url: configuration.httpBaseURL.appendingPathComponent(rawValue)
    )
    request.httpMethod = "GET"
    request.setValue("application/json", forHTTPHeaderField: "Accept")
    return request
  }
}
